build-support/bin/generate_docs.py

```diff
@@ -269,28 +298,57 @@ def do_delete(doc_to_delete):
         for doc in docs:
             do_delete(doc)
 
+        # Partition the scopes into goals and subsystems.
+        goals = {}
+        subsystems = {}
+        for scope, shi in self._help_info["scope_to_help_info"].items():
+            if scope == "":
+                continue  # We handle the global scope separately.
+            if shi["is_goal"]:
+                goals[scope] = shi
+            else:
+                subsystems[scope] = shi
+
         # Create the top-level docs in order.
         self._create(
             parent_doc_id=None,
-            slug="reference-global",
+            slug_suffix="global",
             title="Global",
             body=self._render_body(self._help_info["scope_to_help_info"][""]),
         )
-        self._create(parent_doc_id=None, slug="reference-all-goals", title="Goals", body="")
         self._create(
-            parent_doc_id=None, slug="reference-all-subsystems", title="Subsystems", body=""
+            parent_doc_id=None,
+            slug_suffix="all-goals",
```

Comment:
Why remove the `reference` prefix?

Reply:
It's not removed, it's just injected later.